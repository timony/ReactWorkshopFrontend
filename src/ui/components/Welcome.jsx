import React from 'react';
import PropTypes from 'prop-types';
import Clock from './Clock.jsx';

export class Welcome extends React.Component {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
  }

  componentWillUnmount() {
  }

  render() {
    return (
      <div>Welcome, {this.props.name}, its currently <Clock/></div>
    );
  }
}

Welcome.propTypes = {
  name: PropTypes.string.isRequired
};

export default Welcome;
