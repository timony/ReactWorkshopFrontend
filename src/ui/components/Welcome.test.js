import React from 'react';
import renderer from 'react-test-renderer';
import {Welcome} from './Welcome';

describe('Home dump component', () => {
  it('should render with defaults', () => {
    const component = renderer.create(<Welcome/>);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
