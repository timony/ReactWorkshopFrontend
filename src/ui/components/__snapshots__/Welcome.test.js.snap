// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home dump component should render with defaults 1`] = `
<h2>
  It is 
  12:41:11
</h2>
`;
